File: PHP/Solid Principles/interface_segregation/solution/Customer.php
<?php

class Customer implements SubscriptionNotificationService, UserManager
{
    private $_name;
    private $_email;

    public function __construct($name) {
        $this->_name = $name;
    }

    public function getEmail() {
        return $this->_email;
    }

    public function getName() {
        return $this->_name;
    }

    public function changePassword($customer, $newPassword) {
        echo "Changing password...\n";
    }

    public function subscribeToMSNotifications() {
        echo "Notifying password...\n";
    }

    public function subscribeToNewProductsAvailability() {
        echo "subscribing...\n";
    }

    public function updateUserProfile($customer) {
        echo "updating...\n";
    }
}

File: Python/SOLID/Single-Responsibility/Customer.py
class Customer:
    def __init__(self, name: str, email: str):
        self.name = name
        self.email = email

    def get_name(self) -> str:
        return self.name
    
    def get_email(self) -> str:
        return self.email

    def __str__(self):
        return f"Customer name: {self.name}, Email: {self.email}"